builtin/providers/azurerm/resource_arm_image_test.go

```diff
@@ -0,0 +1,343 @@
+		if err != nil {
+			return fmt.Errorf("Bad: Get on vmClient: %s", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound && shouldExist {
+			return fmt.Errorf("Bad: VM %q (resource group %q) does not exist", vmName, resourceGroup)
+		}
+		if resp.StatusCode != http.StatusNotFound && !shouldExist {
+			return fmt.Errorf("Bad: VM %q (resource group %q) still exists", vmName, resourceGroup)
+		}
+
+		log.Printf("[INFO] testing MANAGED IMAGE VM EXISTS - END.")
+
+		return nil
+	}
+}
+
+func testCheckAzureRMImageDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).diskClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_managed_disk" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Managed Disk still exists: \n%#v", resp.Properties)
+		}
+	}
+
+	return nil
+}
+
+var testAccAzureRMImage_standaloneImage = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%[1]d"
+    location = "West Central US"
+}
+
+resource "azurerm_image" "test" {
+    name = "accteste-%[1]d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+	os_disk_os_type = "Linux"
+	os_disk_os_state = "Generalized"
+	os_disk_blob_uri = "https://terraformdev.blob.core.windows.net/packerimages/ubuntu_plain.vhd"
+	os_disk_size_gb = 30
+	os_disk_caching = "None"
+
+    tags {
+        environment = "acctest"
+        cost-center = "ops"
+    }
+}`
+
+var testAccAzureRMImage_customImage_fromVHD = `
+resource "azurerm_resource_group" "test" {
+    name = "acctestRG-%[1]d"
+    location = "West Central US"
+}
+
+resource "azurerm_virtual_network" "test" {
+    name = "acctvn-%[1]d"
+    address_space = ["10.0.0.0/16"]
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+}
+
+resource "azurerm_subnet" "test" {
+    name = "acctsub-%[1]d"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+    virtual_network_name = "${azurerm_virtual_network.test.name}"
+    address_prefix = "10.0.2.0/24"
+}
+
+resource "azurerm_network_interface" "test" {
+    name = "acctni-%[1]d"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+
+    ip_configuration {
+    	name = "testconfiguration1"
+    	subnet_id = "${azurerm_subnet.test.id}"
+    	private_ip_address_allocation = "dynamic"
+    }
+}
+
+resource "azurerm_image" "test" {
+    name = "accteste"
+    location = "West Central US"
+    resource_group_name = "${azurerm_resource_group.test.name}"
+	os_disk_os_type = "Linux"
+	os_disk_os_state = "Generalized"
+	os_disk_blob_uri = "https://terraformdev.blob.core.windows.net/packerimages/ubuntu_plain.vhd"
```

Comment:
> Note that two acceptance tests currently require subscription-specific information (VHD has to be in a user account or VM must be generalized). This information would need to be provided by the person running the tests - open to ideas on improving this.

We could use a multi-part test here - spinning up a VM first, generalising it, then using the image in third test-case - which would allow this to work in any subscription?